File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/DewVial.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;

public class DewVial extends Equipable {
	{
		name = Locale.get("dew_vial");
		description = Locale.get("dew_vial_desc");
		sprite = "item-dew_vial";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);
		this.owner.healOnEnter = true;
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);
		this.owner.healOnEnter = false;
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/DewVial.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale

class DewVial : Equipable() {
	init {
		name = Locale.get("dew_vial")
		description = Locale.get("dew_vial_desc")
		sprite = "item-dew_vial"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.healOnEnter = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.healOnEnter = false
	}
}